src/goDebugConfiguration.ts

```diff
@@ -80,7 +80,11 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
 			debugConfiguration['dlvLoadConfig'] = dlvConfig['dlvLoadConfig'];
 		}
 
-		debugConfiguration['dlvToolPath'] = getBinPath('dlv');
+		if (!debugConfiguration['dlvToolPath']) {
+			debugConfiguration['dlvToolPath'] = getBinPath('dlv');

```

Comment:
Can you remind me why using `go.alternateTools` wont work here?
```
"go.alternateTools": {
   "dlv": "your tool path to delve"
}
```

Reply:
Copying from our internal conversation.
ok I looked into this, the ability to override dlv via go.alternateTools isn't hooked up...i.e. we don't call `getBinPathWithPreferredGopath()` when resolving the location for dlv....at any rate I don't think this is a great solution as it means that I have modify settings.json and launch.json, plus I can't easily toggle among different configurations as I believe the options in settings.json apply to all configurations (correct me if I'm wrong)

Reply:
> we don't call `getBinPathWithPreferredGopath()` when resolving the location for dlv.

Do you mean we wont end up calling it if using alternateTools or that we shouldnt call it?
 `getBinPath` internally does call `getBinPathWithPreferredGopath`

> at any rate I don't think this is a great solution as it means that I have modify settings.json and launch.json,

we would be modifying just the settings, why launch.json

> I can't easily toggle among different configurations as I believe the options in settings.json apply to all configurations (correct me if I'm wrong)

You are right about that.